astropy/io/fits/tests/test_diff.py

```diff
@@ -579,6 +579,22 @@ def test_diff_nans(self):
         assert np.isnan(diff.diff_values[1][1][0])
         assert diff.diff_values[1][1][1] == 2.0
 
+    def test_diff_types(self):
+        """
+        Regression test for https://github.com/astropy/astropy/issues/4122
+        """
+        
+        f = io.StringIO()
+        
+        a = "1.0"
+        b = 1.0
+        
+        report_diff_values(f, a, b)
+        out = f.getvalue()
+        
+        assert 'a>' in out
+        assert 'b>' in out
```

Comment:
The test looks good, but you might as well extend it to test that the *full* output is what you expect it to be.